docs/conceptual/core/executor.md

```diff
@@ -2,8 +2,147 @@
\ No newline at end of file
+The Substrate runtime is compiled into a native executable and a WebAssembly
+(Wasm) binary.
+
+The native runtime is included as part of the node executable, while the Wasm
+binary is stored on the blockchain under a well known storage key.
+
+These two representations of the runtime may not be the same. For example: when
+the runtime is upgraded. The executor determines which version of the runtime to
+use when dispatching and executing calls.
+
+### Execution Strategy
+
+Before runtime execution begins, the Substrate client chooses which runtime
+execution environment should be used. This is controlled by the execution
+strategy, which can be configured for the different parts of the blockchain
+execution process. The strategies are:
+
+- `Native`: Execute with native build (if available, WebAssembly otherwise).
+- `Wasm`: Only execute with the WebAssembly build.
+- `Both`: Execute with both native (where available) and WebAssembly builds.
+- `NativeElseWasm`: Execute with the native build if possible; if it fails, then
+  execute with WebAssembly.
+
+The default execution strategies for the different parts of the blockchain
+execution process are:
+
+- Syncing: `NativeElseWasm`
+- Block Import: `NativeElseWasm`
+- Block Construction: `Wasm`
+- Off-Chain Worker: `Native`
+- Other: `Native`
+
+### Wasm Execution
+
+The Wasm representation of the Substrate runtime is considered the canonical
+runtime. Because this Wasm runtime is placed in the blockchain storage, the
+network must come to consensus about this binary. Thus it can be verified to be
+consistent across all syncing nodes.
+
+The Wasm execution environment can be more restrictive than the native execution
+environment. For example, the Wasm runtime always executes in a 32-bit
+environment with a 4 GB memory limit.
+
+For these reasons, the blockchain prefers to do block construction with the Wasm
+runtime even though Wasm execution is measurably slower than native execution.
+Wasm execution can help to ensure that block producers create valid blocks.
+
+### Native Execution
+
+The native runtime will only be used by the executor when it is chosen as the
+execution strategy and it is compatible with the requested runtime version. For
+all other execution processes other than block construction, the native runtime
+is preferred since it is more performant. In any situation where the native
+executable should not be run, the canonical Wasm runtime is executed instead.
+
+## Runtime Upgrades
+
+The Substrate runtime is designed to be upgraded so that you can continue to
+iterate on your state transition function even after your blockchain goes live.
+
+### Runtime Versioning
+
+In order for the executor to be able to select the appropriate runtime execution
+environment, it needs to know the `spec_name`, `spec_version` and
+`authoring_version` of both the native and Wasm runtime.
+
+The runtime provides the following versioning properties:
```

Comment:
I would add a link from here to https://github.com/paritytech/substrate/blob/f3a830d3e37d78a869c78c037cc2ac6f0c4e5ddd/node/runtime/src/lib.rs#L79 maybe (it for sure helped me understand better when I peaked at this code)

Reply:
I assume better to do it from crates.parity.io

Reply:
Added a link to here: https://substrate.dev/rustdocs/master/sr_version/struct.RuntimeVersion.html